### PR TITLE
[refactor] - Adjust File Handling Errors

### PR DESCRIPTION
**Pull request by ahrav**
### Description:
This PR address the first point in [this](https://github.com/trufflesecurity/trufflehog/pull/3403#issuecomment-2430039719) comment.

Avoid shadowing `err` by handling it immediately upon encountering an error.
### Checklist:
* [ ] Tests passing (`make test-community`)?
* [ ] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/welcome/install/#local-installation))?

### CODE REVIEW

File: pkg/handlers/ar.go [hunk 1]
```diff
@@ -31,41 +31,35 @@ func (h *arHandler) HandleFile(ctx logContext.Context, input fileReader) (chan [
...
-		defer cancel()
 		defer close(archiveChan)
 
-		// Update the metrics for the file processing.
-		start := time.Now()
-		var err error
-		defer func() {
-			h.measureLatencyAndHandleErrors(start, err)
-			h.metrics.incFilesProcessed()
-		}()
```
Comment (rosecodym) [inline pkg/handlers/ar.go:44]: Does this mean that archives that generate errors during processing are no longer counted by the metric and no longer have their latency measured?

Comment (ahrav) [reply pkg/handlers/ar.go:44]: This change increments `incFilesProcessed` only for Non-Fatal errors, but latency is recorded regardless of success or failure. For Fatal errors, we increment the `incErrors` metric within `measureLatencyAndHandleErrors` instead.

I was concerned when I saw that we’re incrementing the file processed metric even in cases of errors. The metric name may be unclear, as I assumed "processed" meant successfully processed, not just acknowledged. With the current approach, even files we can’t open are counted as processed, which feels somewhat misleading. I’m open to changing it back, though.

File: pkg/handlers/rpm.go [hunk 2]
```diff
@@ -31,48 +31,41 @@ func (h *rpmHandler) HandleFile(ctx logContext.Context, input fileReader) (chan
...
 			}
 		}()
 
-		var rpm *rpmutils.Rpm
-		rpm, err = rpmutils.ReadRpm(input)
+		start := time.Now()
+		rpm, err := rpmutils.ReadRpm(input)
 		if err != nil {
-			ctx.Logger().Error(err, "error reading RPM")
+			ctx.Logger().Error(err, "Error reading rpm file")
```
Comment (mcastorina) [inline pkg/handlers/rpm.go:52]: Are we capitalizing log messages now? Is that a best practice style?

**Review by mcastorina — APPROVED** [review]

File: pkg/handlers/ar.go (hunk 1 above)
Comment (dustin-decker) [reply pkg/handlers/ar.go:44]: Food for thought, the status could be represented with a metric tag and then we could use the same metric visualizations 'by status', which could be "success", "failure", "skipped" etc.

File: pkg/handlers/rpm.go [hunk 3]
```diff
@@ -31,48 +31,41 @@ func (h *rpmHandler) HandleFile(ctx logContext.Context, input fileReader) (chan
...
 			return
 		}
 
-		var reader rpmutils.PayloadReader
-		reader, err = rpm.PayloadReaderExtended()
+		reader, err := rpm.PayloadReaderExtended()
 		if err != nil {
-			ctx.Logger().Error(err, "error getting RPM payload reader")
+			ctx.Logger().Error(err, "Error reading rpm file")
```
Comment (casey-tran) [inline pkg/handlers/rpm.go:58]: Would you want to have the error log string on line 58 to be different than the one located in line 52 in order to make debugging a little easier to find the line that caused the error?

Comment (ahrav) [reply pkg/handlers/rpm.go:58]: Great idea! I’ve been caught reusing the same error message too often. Thanks for pointing that out. 🙇

File: pkg/handlers/rpm.go (hunk 2 above)
Comment (ahrav) [reply pkg/handlers/rpm.go:52]: Oof, nope.

File: pkg/handlers/ar.go (hunk 1 above)
Comment (ahrav) [reply pkg/handlers/ar.go:44]: That's an interesting idea. I'll put together a separate PR with that. Thanks for the suggestion.

**Pull request merged**